internal/handler/employee_handler.go

```diff
@@ -82,3 +102,32 @@ func (e *EmployeeHandler) GetEmployeeById(w http.ResponseWriter, r *http.Request
 
 	response.JSON(w, http.StatusOK, ResponseBody{Data: employeeJSON})
 }
+
+func (e *EmployeeHandler) InsertEmployee(w http.ResponseWriter, r *http.Request) {
+	var newEmployee EmployeeJSON
+	err := request.JSON(r, &newEmployee)
+
+	if err != nil {
+		response.JSON(w, http.StatusUnprocessableEntity, nil)
+		return
+	}
+
+	employee := newEmployee.toEmployeeEntity()
+
+	data, err := e.sv.InsertEmployee(*employee)
+
+	if err != nil && errors.Is(err.(custom_error.CustomError).Err, custom_error.DuplicatedId) {
```

Comment:
Nesse caso se o err for difrente do seu custom ele vai passar pra frente com erro

Reply:
boa, corrigi aqui